common/djangoapps/util/password_policy_validators.py

```diff
@@ -320,3 +320,11 @@ def _validate_password_dictionary(value):
             if edit_distance <= password_max_edit_distance:
                 raise ValidationError(_("Password is too similar to a dictionary word."),
                                       code="dictionary_word")
+
+
+def normalize_password(password):
+    """
+    Normalize all passwords to 'NFKC' across the platform to prevent mismatched hash strings when comparing entered
+    passwords on login. See LEARNER-4283 for more context.
+    """
+    return unicodedata.normalize('NFKC', password)
```

Comment:
Thinking about this, I wonder if it would be helpful to pull out the 'NFKC' into a constant in case it is used in multiple locations and we ever decide to change the normalization metric (I sure hope we do not). I think it could help make some of the tests more clear as well though.

Reply:
Since everything is calling this util function and we only reference 'NFKC'/'NFKD' in the tests I'm not gonna make a constant but if you're really dedicated to that you can do it.